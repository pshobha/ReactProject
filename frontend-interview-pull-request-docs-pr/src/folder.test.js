import React from 'react';
import {shallow} from 'enzyme';
import Folder from './folder.component.js';

describe(`<Folder />`, () => {
	it(`renders a file with file name, date added, and username`, () => {
		const wrapper = shallow(<Folder
			indentationLevel={0}
			folder={{
				id: 'folder1',
				name: 'Folder 1',
				addedBy: {
					id: 'user1',
					name: 'Wilhelm II',
				},
				folders: [
					{
						id: 'folder2',
						name: 'Folder 2',
						folders: [],
						files: [],
						addedBy: {
							id: 'user1',
							name: 'Wilhelm II',
						},
					},
				],
				files: [
					{
						id: 'file1',
						name: 'Genghis-Khan.png',
						dateAdded: 1489183017615,
						addedBy: {
							id: 'user1',
							name: 'Erich Ludendorff',
						}
					},
				],
			}}
		/>);
		expect(wrapper).toMatchSnapshot();
	});
});

