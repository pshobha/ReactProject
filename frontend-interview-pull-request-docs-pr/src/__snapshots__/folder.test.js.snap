// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Folder /> renders a file with file name, date added, and username 1`] = `
<div>
  <div
    className="row"
  >
    <div
      style={
        Object {
          "borderLeft": "0px solid deepskyblue",
        }
      }
    >
      Folder 1
    </div>
    <div>
      3/10/2017
    </div>
    <div>
      Wilhelm II
    </div>
  </div>
  <Folder
    folder={
      Object {
        "addedBy": Object {
          "id": "user1",
          "name": "Wilhelm II",
        },
        "files": Array [],
        "folders": Array [],
        "id": "folder2",
        "name": "Folder 2",
      }
    }
    indentationLevel={1}
  />
  <File
    file={
      Object {
        "addedBy": Object {
          "id": "user1",
          "name": "Erich Ludendorff",
        },
        "dateAdded": 1489183017615,
        "id": "file1",
        "name": "Genghis-Khan.png",
      }
    }
    indentationLevel={1}
  />
</div>
`;
